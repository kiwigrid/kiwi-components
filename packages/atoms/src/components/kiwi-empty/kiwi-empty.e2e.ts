import { newE2EPage } from '@stencil/core/testing';

describe('kiwi-empty', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<kiwi-empty></kiwi-empty>');

    const element = await page.find('kiwi-empty');
    expect(element).toHaveClass('hydrated');
  });
});
