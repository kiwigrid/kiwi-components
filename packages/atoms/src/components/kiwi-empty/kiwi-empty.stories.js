import { html } from 'lit-html';
import notes from './readme.md';

export default {
  title: 'KiwiComponents/Empty',
  parameters: { docs: { description: { component: notes } } },
  decorators: [],
};

export const withContent = (args) => {
  const { content, glyphicon, size } = args;

  return html` <div class="panel panel-default">
    <div class="panel-body">
      <kiwi-empty with-content="true" size="${size}" glyphicon="${glyphicon}">
        ${content}
      </kiwi-empty>
    </div>
  </div>`;
};

export const withoutContent = (args) => {
  const { glyphicon, size } = args;

  return html` <div class="panel panel-default">
    <div class="panel-body">
      <kiwi-empty with-content="false" size="${size}" glyphicon="${glyphicon}">
      </kiwi-empty>
    </div>
  </div>`;
};
withoutContent.args = withContent.args = {
  glyphicon: 'refresh',
};
withContent.args = { ...withContent.args, content: 'Connecting...' };
withContent.argTypes = withoutContent.argTypes = {
  size: {
    control: {
      type: 'select',
      options: ['small', 'medium', 'large'],
    },
    defaultValue: 'medium',
  },
};
