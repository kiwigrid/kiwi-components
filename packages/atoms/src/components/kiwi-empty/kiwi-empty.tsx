import { Component, ComponentInterface, h, JSX, Prop } from '@stencil/core';

export type Size = 'small' | 'medium' | 'large';

const SIZES: { [Key in Size]: string } = {
  small: '150%',
  medium: '300%',
  large: '600%',
};

@Component({
  tag: 'kiwi-empty',
  shadow: false,
})
export class KiwiEmpty implements ComponentInterface {
  /**
   * Show optional content below the icon.
   */
  @Prop()
  withContent = false;

  /**
   * The size of the icon.
   */
  @Prop()
  size: Size = 'large';

  /**
   * The name of the glyphicon icon to use.
   * So if you want to render a `glyphicon-ok` pass `ok` as the name.
   */
  @Prop()
  glyphicon = 'inbox';

  /**
   * Provide a custom class to use instead of glyphicons.
   */
  @Prop()
  icon?: string;

  /** Additional css class to apply to the container. */
  @Prop()
  containerClass = '';

  render(): JSX.Element {
    const iconClass = this.icon
      ? this.icon
      : `glyphicon glyphicon-${this.glyphicon}`;
    return (
      <div
        class={`text-center mt-3 ${this.containerClass}`}
        style={{ color: '#9d9d9d' }}
      >
        <span class={iconClass} style={{ fontSize: SIZES[this.size] }}></span>
        {this.withContent && (
          <p class="mt-2">
            <slot />
          </p>
        )}
      </div>
    );
  }
}
