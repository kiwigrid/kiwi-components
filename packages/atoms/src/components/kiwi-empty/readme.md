# kiwi-empty

<!-- Auto Generated Below -->


## Properties

| Property         | Attribute         | Description                                                                                             | Type                             | Default     |
| ---------------- | ----------------- | ------------------------------------------------------------------------------------------------------- | -------------------------------- | ----------- |
| `containerClass` | `container-class` | Additional css class to apply to the container.                                                         | `string`                         | `''`        |
| `glyphicon`      | `glyphicon`       | The name of the glyphicon icon to use. So if you want to render a `glyphicon-ok` pass `ok` as the name. | `string`                         | `'inbox'`   |
| `icon`           | `icon`            | Provide a custom class to use instead of glyphicons.                                                    | `string \| undefined`            | `undefined` |
| `size`           | `size`            | The size of the icon.                                                                                   | `"large" \| "medium" \| "small"` | `'large'`   |
| `withContent`    | `with-content`    | Show optional content below the icon.                                                                   | `boolean`                        | `false`     |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
