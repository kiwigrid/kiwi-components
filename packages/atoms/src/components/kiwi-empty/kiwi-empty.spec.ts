import { newSpecPage } from '@stencil/core/testing';
import { KiwiEmpty } from './kiwi-empty';

describe('kiwi-empty', () => {
  it('builds', () => {
    expect(new KiwiEmpty()).toBeTruthy();
  });

  // [
  //   ['default', { data: [] }],
  //   ['content', { data: ['with-content', 'true'], content: 'No data' }],
  //   ['glyphicon suffix', { data: ['glyphicon', 'kittens'] }],
  //   ['custom class', { data: ['icon', 'icon icon-beer'] }],
  //   ['size: small', { data: ['size', 'small'] }],
  //   ['size: medium', { data: ['size', 'medium'] }],
  //   ['size: large', { data: ['size', 'large'] }],
  // ].forEach(
  //   ([
  //     title,
  //     {
  //       data: [prop, value],
  //       content,
  //     },
  //   ]: [string, { data: [string, string]; content?: string }]) => {
  //     it(`renders ${title}`, async () => {
  //       const empty = await newSpecPage({
  //         components: [KiwiEmpty],
  //         html: content
  //           ? `<kiwi-empty ${prop}="${value}">${content}</kiwi-empty>`
  //           : `<kiwi-empty ${prop}="${value}" />`,
  //       });

  //       expect(empty.root).toMatchSnapshot();
  //     });
  //   },
  // );

  it.each([
    ['default', { data: [] }],
    ['content', { data: ['with-content', 'true'], content: 'No data' }],
    ['glyphicon suffix', { data: ['glyphicon', 'kittens'] }],
    ['custom class', { data: ['icon', 'icon icon-beer'] }],
    ['size: small', { data: ['size', 'small'] }],
    ['size: medium', { data: ['size', 'medium'] }],
    ['size: large', { data: ['size', 'large'] }],
  ])('renders %s', async (title, args) => {
    const prop = args.data[0];
    const value = args.data[1];
    const content = (args as { data: string[]; content: string }).content;

    const empty = await newSpecPage({
      components: [KiwiEmpty],
      html: content
        ? `<kiwi-empty ${prop}="${value}">${content}</kiwi-empty>`
        : `<kiwi-empty ${prop}="${value}" />`,
    });

    expect(empty.root).toMatchSnapshot();
  });
});
