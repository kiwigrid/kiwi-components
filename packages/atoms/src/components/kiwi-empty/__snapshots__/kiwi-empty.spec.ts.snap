// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kiwi-empty renders content 1`] = `
<kiwi-empty with-content="true">
  <!---->
  <div class="mt-3 text-center" style="color: #9d9d9d;">
    <span class="glyphicon glyphicon-inbox" style="font-size: 600%;"></span>
    <p class="mt-2">
      No data
    </p>
  </div>
</kiwi-empty>
`;

exports[`kiwi-empty renders custom class 1`] = `
<kiwi-empty icon="icon icon-beer">
  <!---->
  <div class="mt-3 text-center" style="color: #9d9d9d;">
    <span class="icon icon-beer" style="font-size: 600%;"></span>
  </div>
</kiwi-empty>
`;

exports[`kiwi-empty renders default 1`] = `
<kiwi-empty undefined="undefined">
  <!---->
  <div class="mt-3 text-center" style="color: #9d9d9d;">
    <span class="glyphicon glyphicon-inbox" style="font-size: 600%;"></span>
  </div>
</kiwi-empty>
`;

exports[`kiwi-empty renders glyphicon suffix 1`] = `
<kiwi-empty glyphicon="kittens">
  <!---->
  <div class="mt-3 text-center" style="color: #9d9d9d;">
    <span class="glyphicon glyphicon-kittens" style="font-size: 600%;"></span>
  </div>
</kiwi-empty>
`;

exports[`kiwi-empty renders size: large 1`] = `
<kiwi-empty size="large">
  <!---->
  <div class="mt-3 text-center" style="color: #9d9d9d;">
    <span class="glyphicon glyphicon-inbox" style="font-size: 600%;"></span>
  </div>
</kiwi-empty>
`;

exports[`kiwi-empty renders size: medium 1`] = `
<kiwi-empty size="medium">
  <!---->
  <div class="mt-3 text-center" style="color: #9d9d9d;">
    <span class="glyphicon glyphicon-inbox" style="font-size: 300%;"></span>
  </div>
</kiwi-empty>
`;

exports[`kiwi-empty renders size: small 1`] = `
<kiwi-empty size="small">
  <!---->
  <div class="mt-3 text-center" style="color: #9d9d9d;">
    <span class="glyphicon glyphicon-inbox" style="font-size: 150%;"></span>
  </div>
</kiwi-empty>
`;
